pubsub/batcher/batcher.go

```diff
@@ -283,5 +298,13 @@ func (b *Batcher) Shutdown() {
 	b.mu.Lock()
 	b.shutdown = true
 	b.mu.Unlock()
+
+	// On shutdown, ensure that we attempt to flush any pending items
+	// if there's a minimum batch size.
+	if atomic.LoadInt32(&b.nHandlers) < int32(b.opts.MaxHandlers) {
```

Comment:
It shouldn't be necessary to use `atomic` to check/modify nHandlers. There's a mutex on the struct to protect it. IIUC, the problem is that you added a read on this line outside of the lock. I think if you move the b.mu.Unlock a few lines up to below this new stanza, it will be fine.